src/Controller.ts

```diff
@@ -22,6 +24,49 @@ type Bindings = Partial<{ [eventName in ReactEventHandlerKey]: Fn[] }>
  * @template BinderType the type the bind function should return
  */
 export default class Controller {
+  constructor(private classes: RecognizerClass[]) {}
+
+  public bind = (...args: any[]) => {
+    this.resetBindings()
+    for (let RecognizerClass of this.classes) {
+      new RecognizerClass(this, args).addBindings()
+    }
+
+    // we also add event bindings for native handlers
+    if (this.nativeRefs) {
+      for (let eventName in this.nativeRefs)
+        this.addBindings(
+          eventName as ReactEventHandlerKey,
+          // @ts-ignore we're cheating when it comes to event type :(
+          this.nativeRefs[eventName] as Fn
+        )
+    }
+
+    // If config.domTarget is set we add event listeners to it and return the clean function.
+    if (this.isDomTargetDefined) return
+    // If not, we return an object that contains gesture handlers mapped to react handler event keys.
+
+    const bindings: ReactEventHandlers = {}
+    const { captureString } = this.config
+
+    Object.entries(this.bindings).forEach(([event, fns]) => {
+      const fnsArray = Array.isArray(fns) ? fns : [fns]
+      const key = (event + captureString) as ReactEventHandlerKey
+      bindings[key] = chainFns(...(fnsArray as Fn[]))
+    })
+
+    return bindings
+  }
+
+  public effect = () => {
+    if (this.isDomTargetDefined) {
+      this.bind()
+      this.addDomTargetListeners()
+    }
+    return this!.clean
```

Comment:
@thephoenixofthevoid 
Why is the `!` needed? Removing it doesn't throw any error?